tests/EffectiveCSharp.Analyzers.Tests/PreferDeclarationInitializersToAssignmentStatementsTests.cs

```diff
@@ -0,0 +1,797 @@
+              }
+            }
+            """,
+            ReferenceAssemblyCatalog.Latest);
+    }
+
+    [Fact]
+    public async Task FieldsInitializedWithPropertySetters()
+    {
+        // Fields are initialized within property setters rater than directly or with ctors
+        // The analyzer should recognize this case and should not flag for declaration initialization
+        await Verifier.VerifyAnalyzerAsync(
+            """
+            public class MyClass
+            {
+              private string label;
+              
+              public string Label
+              {
+                get => label;
+                set => label = value;
+              }
+            }
+            """,
+            ReferenceAssemblyCatalog.Latest);
+    }
+
+    [Fact]
+    public async Task FieldsWithCtorInitializationPartialClass()
+    {
+        // Fields in partial classes that are initialized in different parts of the class.
+        // Ensure the analyzer correctly handles partial classes and fields initialized across them.
+        await Verifier.VerifyAnalyzerAsync(
+            """
+            public partial class MyClass
+            {
+              {|ECS1203:private string label;|}
+            }
+            
+            public partial class MyClass
+            {  
+              public MyClass()
+              {
+                label = "Initialized in partial class ctor";
+              }
+            }
+            """,
+            ReferenceAssemblyCatalog.Latest);
+    }
+
+    [Fact]
+    public async Task FieldsWithNoInitialization()
+    {
+        // Fields are not initialized either in the declaration or the ctor
+        // The analyzer suggests adding an initializer in such cases
+        await Verifier.VerifyAnalyzerAsync(
+            """
+            public class MyClass
+            {
+              {|ECS1203:private string label;|}
+              
+              public MyClass()
+              {
+              }
+            }
+            """,
+            ReferenceAssemblyCatalog.Latest);
+    }
+
+    [Fact]
+    public async Task FieldsInitializedBasedOnExternalParametersOrData()
+    {
+        await Verifier.VerifyAnalyzerAsync(
+            """
+            public class MyClass
+            {
+              private string label;
+              
+              public MyClass(string label)
+              {
+                this.label = label;
+              }
+            }
+            """,
+            ReferenceAssemblyCatalog.Latest);
+    }
+
+    [Fact]
+    public async Task InitializationWithCtorOverloadChains()
+    {
+        await Verifier.VerifyAnalyzerAsync(
+            """
+            public class MyClass
+            {
+              private string label;
+              
+              public MyClass() : this("Default Label")
+              {
+              }
+              
+              public MyClass(string label)
+              {
+                {|ECS1203:this.label = label;|}
```

Comment:
@rjmurillo I don't know if this is a situation we should flag. To me, this is an issue as to whether it is acceptable to pass that hardcoded value rather than just write label = "Default Label"; inside of the first constructor.

I say this because in my eyes, the moment you use a parameter in a constructor, we cannot tell you to initialize in the declaration. In the IL, the declaration initialization happens first, then the assignment in the constructor. That to me goes against the rule exception of not initializing to zero or null. Now, this isn't the same situation, but we do not initialize to zero or null because it would be redundant, as is the case with this situation.

I don't think the analyzer should flag anything in this test case. 

Item 12 from the book: 

The second inefficiency comes when you create multiple initializations
for the same object. You should use the initializer syntax only for vari-
ables that receive the same initialization in all constructors.

I think item 14 also mentions something about constructor chaining being preferred over copy pasting initializations

IL example:

        IL_0000: ldarg.0
        IL_0001: ldstr "lol"
        IL_0006: stfld string C::lol
        IL_000b: ldarg.0
        IL_000c: call instance void [System.Runtime]System.Object::.ctor()
        IL_0011: ldarg.0
        IL_0012: ldarg.1
        IL_0013: stfld string C::lol
        IL_0018: ret

Reply:
> You should use the initializer syntax only for variables that receive the same initialization in all constructors.

I missed that nuance. The annotation should be removed as a positive case